src/js/player.js

```diff
@@ -2679,57 +2679,138 @@ class Player extends Component {
+   *
+   * @listens focus
+   */
+  handleFocus(event) {
+    Events.on(document, 'keydown', Fn.bind(this, this.handleKeyPress));
+  }
+
+  /**
+   * Called when a `Player` loses focus. Turns off the listener for
+   * `keydown` events. Which Stops `this.handleKeyPress` from getting called.
+   *
+   * @param {EventTarget~Event} event
+   *        The `blur` event that caused this function to be called.
+   *
+   * @listens blur
+   */
+  handleBlur(event) {
+    Events.off(document, 'keydown', Fn.bind(this, this.handleKeyPress));
+  }
+
+  /**
+   * Called when this Player has focus and a key gets pressed down, or when
+   * any Component of this player receives a key press that it doesn't handle.
+   * This allows player-wide hotkeys (either as defined below, or optionally
+   * by an external function).
+   *
+   * @param {EventTarget~Event} event
+   *        The `keydown` event that caused this function to be called.
+   *
+   * @listens keydown
    */
   handleKeyPress(event) {
 
-    if (this.options_.hotkeys !== false) {
+    if (this.options_.userActions && this.options_.userActions.hotkeys && (this.options_.userActions.hotkeys !== false)) {
 
-      if (typeof this.options_.hotkeys === 'function') {
+      if (typeof this.options_.userActions.hotkeys === 'function') {
 
-        this.options_.hotkeys(event);
+        this.options_.userActions.hotkeys.call(this, event);
 
-      // Trigger the ControlBar equivalent for each key, if possible
-      } else if (keycode.isEventKey(event, 'f')) {
-        // "f" = toggle fullscreen
+      } else {
 
-        event.preventDefault();
+        this.handleHotkeys(event);
 
-        const cb = this.player_.controlBar;
-        const fullscreenToggle = cb && cb.fullscreenToggle;
+      }
+    }
+  }
 
-        if (fullscreenToggle && fullscreenToggle.handleClick) {
-          fullscreenToggle.handleClick();
-        }
-      } else if (keycode.isEventKey(event, 'm')) {
-        // "m" = toggle mute
+  /**
+   * Called when this Player receives a hotkey keydown event.
+   * Supported player-wide hotkeys are:
+   *
+   *   f          - toggle fullscreen
+   *   m          - toggle mute
+   *   k or Space - toggle play/pause
+   *
+   * @param {EventTarget~Event} event
+   *        The `keydown` event that caused this function to be called.
+   */
+  handleHotkeys(event) {
+    const hotkeys = this.options_.userActions ? this.options_.userActions.hotkeys : null;
+    let fullscreenKey;
+    let muteKey;
+    let playPauseKey;
 
-        event.preventDefault();
+    if (hotkeys) {
+      fullscreenKey = hotkeys.fullscreenKey;
+      muteKey = hotkeys.muteKey;
+      playPauseKey = hotkeys.playPauseKey;
+    }
 
-        const cb = this.player_.controlBar;
-        const volumePanel = cb && cb.volumePanel;
-        const muteToggle = (cb && cb.muteToggle) || (volumePanel && volumePanel.muteToggle);
+    if (typeof fullscreenKey !== 'function') {
```

Comment:
should we respect a falsy value here to disable that hotkey?

Reply:
Maybe in the future, but for now the `options.userActions.hotkeys` is one type of override (the same as `userActions.doubleClick`), that can take `true`, `false` or an override `function`, whereas `fullscreenKey` is modeled after the option in https://github.com/ctd1500/videojs-hotkeys, which is *only* a function.

I realize that the `options.userActions` got pretty complicated with essentially two different kinds of overrides; I did my best to document it in `options.md`, but I can cut some of it out if it makes more sense?